public/app/plugins/datasource/cloud-monitoring/components/ConfigEditor/JWTConfig.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { FormEvent, useState } from 'react';
+import { startCase, isObject } from 'lodash';
+import { Button, FileUpload, InlineField, Input, useStyles, Alert } from '@grafana/ui';
+import { css, cx } from '@emotion/css';
+import { GrafanaTheme } from '@grafana/data';
+
+const configKeys = [
+  'type',
+  'project_id',
+  'private_key_id',
+  'private_key',
+  'client_email',
+  'client_id',
+  'auth_uri',
+  'token_uri',
+  'auth_provider_x509_cert_url',
+  'client_x509_cert_url',
+];
+
+export interface JWT {
+  token_uri: string;
+  client_email: string;
+  private_key: string;
+  project_id: string;
+}
+
+export interface Props {
+  onChange: (jwt: JWT) => void;
+  isConfigured: boolean;
+}
+
+const validateJson = (json: { [key: string]: string }) => isObject(json) && configKeys.every((key) => !!json[key]);
+
+export function JWTConfig({ onChange, isConfigured }: Props) {
+  const styles = useStyles(getStyles);
+  const [enableUpload, setEnableUpload] = useState<boolean>(!isConfigured);
+  const [error, setError] = useState<string | null>(null);
+
+  return enableUpload ? (
+    <>
+      <FileUpload
+        className={styles}
+        accept="application/json"
+        onFileUpload={(event: FormEvent<HTMLInputElement>) => {
+          if (event?.currentTarget?.files?.length === 1) {
+            setError(null);
+            const reader = new FileReader();
+            const readerOnLoad = () => {
+              return (e: any) => {
+                const json = JSON.parse(e.target.result);
+                if (validateJson(json)) {
+                  onChange(json as JWT);
```

Comment:
I guess if you take my suggestion to return something to the onchange, then you'd want to have a try/catch here instead of just the if statement